src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/Maintainability/CodeMetricsComputationTests.cs

```diff
@@ -550,6 +551,124 @@ class C2 : C1 { }
             VerifyCSharp(source, expectedMetricsText);
         }
 
+        [Fact, WorkItem(2133, "https://github.com/dotnet/roslyn-analyzers/issues/2133")]
+        public void MethodWithAnonymousType()
+        {
+            var source = @"
+class C
+{
+    object M1(C1 c)
+    {
+        return new {a = 1};
+    }
+}
+";
+
+            var expectedMetricsText = @"
+Assembly: (Lines: 8, ExecutableLines: 1, MntIndex: 94, CycCxty: 1, CoupledTypes: {C1}, DepthInherit: 1)
+   C: (Lines: 8, ExecutableLines: 1, MntIndex: 94, CycCxty: 1, CoupledTypes: {C1}, DepthInherit: 1)
+      C.M1(C1): (Lines: 4, ExecutableLines: 1, MntIndex: 94, CycCxty: 1, CoupledTypes: {C1})
+";
+
+            VerifyCSharp(source, expectedMetricsText, true);
+        }
+
+        [Fact, WorkItem(2133, "https://github.com/dotnet/roslyn-analyzers/issues/2133")]
+        public void MethodWithGenericTypes()
+        {
+            var source = @"
+class C
+{
+    void M1()
+    {
+        G<int> a = null;
+        G<string> b = null;
+        G<C1> c = null;
+        G<G<C2>> d = null;
+    }
+}
+class G<T> {}
+";
+
+            var expectedMetricsText = @"
+Assembly: (Lines: 12, ExecutableLines: 4, MntIndex: 88, CycCxty: 2, CoupledTypes: {C1, C2, G<>}, DepthInherit: 1)
+   C: (Lines: 11, ExecutableLines: 4, MntIndex: 77, CycCxty: 1, CoupledTypes: {C1, C2, G<>}, DepthInherit: 1)
+      C.M1(): (Lines: 7, ExecutableLines: 4, MntIndex: 77, CycCxty: 1, CoupledTypes: {C1, C2, G<>})
+   G<T>: (Lines: 1, ExecutableLines: 0, MntIndex: 100, CycCxty: 1, DepthInherit: 1)
+";
+
+            VerifyCSharp(source, expectedMetricsText, true);
```

Comment:
@mavasani Looks like this test class wasn't updated to use the new test framework. `C1` and `C2` are not declared yet there is no compiler error.

Reply:
The `true` being passed as the final argument to VerifyCSharp seems to indicate that we expect compiler errors to happen. The analyzer still appears to be able to produce meaningful information, even though it sees some of the types as having type errors. I was following the patterns that I saw in nearby methods, but if you'd prefer I can add all the necessary classes to prevent the test from failing.

Reply:
Up to you but that's fine like this I think. We have switch to a different "test framework" so I thought that might have some parts broken hecne this comment.

Reply:
I agree with @Evangelink - I think it seems best to not have compile errors for this test. If you wish to also test the functionality in presence of compiler errors, please add another test with appropriate test name.

Reply:
Fixed compile errors on all the test methods I added. I also made one change to the base class to make it easier to track down unexpected diagnostic messages in case of failure.